browser/sources.gni

```diff
@@ -360,6 +360,8 @@ if (is_mac) {
     "//brave/browser/mac/keystone_glue.mm",
     "//brave/browser/mac/keystone_registration.h",
     "//brave/browser/mac/keystone_registration.mm",
+    "//brave/browser/upgrade_detector/upgrade_when_idle.cc",
```

Comment:
put this in its own BUILD.gn and add this to `brave_chrome_browser_allow_circular_includes_from`

Reply:
I extracted the `BUILD.gn` file in 7cfbeb1e3599551318ccd061dd72605f5fd1da75 and renamed the `upgrade_detector/` subdir to `upgrade_when_idle/` in 529162f34644ba8bb06c45464aa8451d9f60427e.